services/movie_session.py

```diff
@@ -0,0 +1,58 @@
+from django.core.exceptions import ObjectDoesNotExist
+from db.models import MovieSession, CinemaHall, Movie
+from datetime import date, datetime
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int,
+) -> MovieSession:
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie_id=movie_id,
+        cinema_hall_id=cinema_hall_id,
+    )
+
+
+def get_movies_sessions(session_date: str = "") -> MovieSession | None:
+    if session_date:
+        year, month, day = session_date.split("-")
+        return MovieSession.objects.filter(
+            show_time__date=date(int(year), int(month), int(day)),
+        )
+    return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: datetime = None,
+        movie_id: int = None,
+        cinema_hall_id: int = None
+) -> None:
+    try:
+        movie_session = MovieSession.objects.get(id=session_id)
+    except ObjectDoesNotExist:
+        print(f"Movie Session with ID {session_id} does not exist")
+    else:
+        if show_time:
+            movie_session.show_time = show_time
+        if cinema_hall_id:

```

Comment:
Can you refactor this so that you will not need to get movie session if none of the update parameters has been passed?

Reply:
Try a different approach:
1. create a dictionary, 
2.  check if any of the optional arguments are passed to this function. In case if there are - you will add them to the dict. 
3. In the end check if your dict has some values to update and if yes you will call update on queryset, and if there are not any you don't do anything